iterative/aws/provider.go

```diff
@@ -62,6 +62,32 @@ func ResourceMachineCreate(ctx context.Context, d *schema.ResourceData, m interf
 	}
 	svc := ec2.NewFromConfig(config)
 
+	// Availability zone
+	if availabilityZone == "" {
+		offeringsInput := &ec2.DescribeInstanceTypeOfferingsInput{
+			LocationType: types.LocationTypeAvailabilityZone,
+			Filters: []types.Filter{
+				{
+					Name:   aws.String("instance-type"),
+					Values: []string{instanceType},
+				},
+			},
+		}
+
+	loop:
+		for offeringsPaginator := ec2.NewDescribeInstanceTypeOfferingsPaginator(svc, offeringsInput); offeringsPaginator.HasMorePages(); {
+			page, err := offeringsPaginator.NextPage(ctx)
+			if err != nil {
+				return err
+			}
+
+			for _, offering := range page.InstanceTypeOfferings {
```

Comment:
Do we need this loop here?

Reply:
Alternatively, we can write the same code like this:

```go
if len(page.InstanceTypeOfferings) > 0 {
        availabilityZone = aws.ToString(page.InstanceTypeOfferings[0].Location)
        break
}
```

Reply:
@tasdomas, would you prefer the latter?

Reply:
I would - it's a bit more explicit, but I'm fine with the PR as it is.

Reply:
Resolved with https://github.com/iterative/terraform-provider-iterative/pull/670/commits/6df5da68d3c7e480680b93a5137a30a65b09ff7f

Reply:
@tasdomas, now “dr[y]er” with https://github.com/iterative/terraform-provider-iterative/pull/670/commits/8efff82ef80abd7e0574e0d8e1162675108e7ebf